### PR TITLE
TW-482: Migration go route for web

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #482 

- Resolved:

### MOBILE

https://github.com/linagora/twake-on-matrix/assets/99852347/5380c6fe-3b6f-4052-a500-c6955daa37d9

### WEB

### CODE REVIEW

File: lib/pages/chat_list/chat_list.dart [hunk 1]
```diff
@@ -49,14 +49,28 @@ enum PopupMenuAction {
...
+  groups,
+  messages,
+  spaces,
+}
 
 class ChatList extends StatefulWidget {
   static BuildContext? contextForVoip;
 
-  final Widget? child;
+  final String? activeChat;
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:62]: what is it?

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import 'package:fluffychat/widgets/layouts/adaptive_layout/adaptive_scaffold.dart';
+import 'package:fluffychat/widgets/layouts/enum/adaptive_destinations_enum.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+class AppScaffoldView extends StatelessWidget {
+  final AdaptiveScaffoldAppController controller;
+
+  final String? activeChat;
```
Comment (hoangdat) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart:16]: what is it?

File: lib/pages/search/search.dart [hunk 3]
```diff
@@ -18,7 +18,8 @@ import 'package:go_router/go_router.dart';
 import 'package:matrix/matrix.dart';
 
 class Search extends StatefulWidget {
-  const Search({super.key});
+  final Function()? callBack;
```
Comment (drminh2807) [inline lib/pages/search/search.dart:21]:
```suggestion
  final Function()? callback;
```

Comment (sherlockvn) [reply lib/pages/search/search.dart:21]: `void final Function()? callback;`

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_route.dart [hunk 4]
```diff
@@ -19,13 +20,14 @@ class AdaptiveScaffoldRoute extends StatelessWidget {
...
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       backgroundColor: Theme.of(context).colorScheme.surface,
-      appBar: const AdaptiveScaffoldAppBar(),
+      appBar: displayAppBar == true ? const AdaptiveScaffoldAppBar() : null,
       body: AdaptiveLayout(
         internalAnimations: false,
         body: SlotLayout(
```
Comment (sherlockvn) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold_route.dart:33]: `displayAppBar ? const AdaptiveScaffoldAppBar() : null`

Comment (nqhhdev) [reply lib/widgets/layouts/adaptive_layout/adaptive_scaffold_route.dart]:
```
 final bool? displayAppBar;
```

Try checking that the value isn't 'null' before using it as a condition.

File: lib/pages/chat/chat_view_style.dart [hunk 5]
```diff
@@ -7,18 +7,15 @@ class ChatViewStyle {
 
   static double get paddingVerticalActionButtons => 8.0;
 
-  static double toolbarHeight(BuildContext context) =>
-      responsive.isMobile(context) ? 104 : 86;
+  static double toolbarHeight(BuildContext context) => 56;
```
Comment (sherlockvn) [inline lib/pages/chat/chat_view_style.dart:10]: remove context, and use const for it

File: lib/pages/chat_details/chat_detail_view_style.dart [hunk 6]
```diff
@@ -1,15 +1,5 @@
...
-  static ResponsiveUtils responsive = getIt.get<ResponsiveUtils>();
-
-  static double toolbarHeight(BuildContext context) =>
-      responsive.isMobile(context) ? 96 : 56;
-
-  static EdgeInsetsDirectional paddingTitle(BuildContext context) =>
-      responsive.isMobile(context)
-          ? const EdgeInsetsDirectional.only(top: 40)
-          : EdgeInsetsDirectional.zero;
+  static double toolbarHeight(BuildContext context) => 56;
```
Comment (sherlockvn) [inline lib/pages/chat_details/chat_detail_view_style.dart:4]: same

File: lib/pages/chat_details/chat_details.dart [hunk 7]
```diff
@@ -20,7 +20,8 @@ import 'package:fluffychat/widgets/matrix.dart';
 enum AliasActions { copy, delete, setCanonical }
 
 class ChatDetails extends StatefulWidget {
-  const ChatDetails({Key? key}) : super(key: key);
+  final String roomId;
+  const ChatDetails({Key? key, required this.roomId}) : super(key: key);
```
Comment (sherlockvn) [inline lib/pages/chat_details/chat_details.dart:24]: `const ChatDetails({super.key, required this.roomId});`

File: lib/pages/chat_draft/draft_chat_view.dart [hunk 8]
```diff
@@ -38,25 +38,22 @@ class DraftChatView extends StatelessWidget {
...
+          children: [
+            TwakeIconButton(
+              tooltip: L10n.of(context)!.back,
+              icon: Icons.arrow_back,
+              onPressed: () => context.pop(),
+              paddingAll: 8.0,
+              margin: const EdgeInsets.symmetric(vertical: 12.0),
+            ),
+            Expanded(
+              child: _EmptyChatTitle(
```
Comment (sherlockvn) [inline lib/pages/chat_draft/draft_chat_view.dart:51]: don't use controller inside StatelessWidget

File: lib/pages/chat_draft/draft_chat_view_style.dart [hunk 9]
```diff
@@ -5,11 +5,5 @@ import 'package:flutter/material.dart';
...
   static ResponsiveUtils responsive = getIt.get<ResponsiveUtils>();
 
-  static double toolbarHeight(BuildContext context) =>
-      responsive.isMobile(context) ? 104 : 86;
-
-  static EdgeInsetsDirectional paddingTopScreen(BuildContext context) =>
-      EdgeInsetsDirectional.only(
-        top: responsive.isMobile(context) ? 40 : 0,
-      );
+  static double toolbarHeight(BuildContext context) => 56;
```
Comment (sherlockvn) [inline lib/pages/chat_draft/draft_chat_view_style.dart:8]: same

File: lib/pages/chat_draft/draft_chat_view.dart (hunk 8 above)
Comment (nqhhdev) [reply lib/pages/chat_draft/draft_chat_view.dart:51]: Written by `sherlockvn` 😁

File: lib/pages/search/search.dart (hunk 3 above)
Comment (nqhhdev) [reply lib/pages/search/search.dart]: Fixed

File: lib/pages/chat/chat_view_style.dart (hunk 5 above)
Comment (nqhhdev) [reply lib/pages/chat/chat_view_style.dart]: Fixed

File: lib/pages/chat_details/chat_details.dart (hunk 7 above)
Comment (nqhhdev) [reply lib/pages/chat_details/chat_details.dart]: Fixed

File: lib/pages/chat_draft/draft_chat_view.dart (hunk 8 above)
Comment (nqhhdev) [reply lib/pages/chat_draft/draft_chat_view.dart:54]: - Removed `controller` inside stl, I move it to `argument` pls check it

File: lib/pages/chat_list/chat_list.dart (hunk 1 above)
Comment (hoangdat) [reply lib/pages/chat_list/chat_list.dart:62]:
```suggestion
  final String? activeRoomId;
```

File: lib/pages/chat_list/chat_list.dart [hunk 10]
```diff
@@ -49,14 +49,28 @@ enum PopupMenuAction {
...
+
+  final Widget? bottomNavigationBar;
 
-  const ChatList({Key? key, this.child}) : super(key: key);
+  final Function()? onTapSearch;
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:66]: Can you explain a little bit?

File: lib/pages/chat_list/chat_list_header.dart [hunk 11]
```diff
@@ -4,161 +4,158 @@ import 'package:fluffychat/widgets/matrix.dart';
 import 'package:fluffychat/widgets/twake_components/twake_header.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
-import 'package:go_router/go_router.dart';
 
 class ChatListHeader extends StatelessWidget {
   final ChatListController controller;
+  final Function()? onTapSearch;
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list_header.dart:10]: jdem

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart (hunk 2 above)
Comment (hoangdat) [reply lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart:24]: idem?

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:matrix/matrix.dart';
+
+class AppScaffoldView extends StatelessWidget {
+  final ValueNotifier<AdaptiveDestinationEnum> activeNavigationBar;
+  final PageController pageController;
+  final Future<Profile?> fetchOwnProfile;
+  final int activeNavigationBarIndex;
+  final OnTapSearch onTapSearch;
```
Comment (hoangdat) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold_view.dart:19]: why need to see it every where?

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold.dart [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+import 'package:go_router/go_router.dart';
+import 'package:matrix/matrix.dart';
+
+typedef OnTapSearch = Function();
+typedef OnCallbackFromSearch = Function();
+typedef OnClientSelectedSetting = Function(Object object, BuildContext context);
+typedef OnDestinationSelected = Function(int index);
+
+class AdaptiveScaffoldApp extends StatefulWidget {
+  final String? activeChat;
```
Comment (hoangdat) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold.dart:16]: idem. If it is `id`, please keep the `id`

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_primary_navigation.dart [hunk 14]
```diff
@@ -2,24 +2,25 @@ import 'package:fluffychat/pages/chat_list/client_chooser_button.dart';
...
 import 'package:fluffychat/widgets/layouts/adaptive_layout/adaptive_scaffold_primary_navigation_style.dart';
-import 'package:fluffychat/widgets/layouts/adaptive_layout/adaptive_scaffold_shell.dart';
 import 'package:fluffychat/widgets/matrix.dart';
 import 'package:flutter/material.dart';
 import 'package:matrix/matrix.dart';
 import 'package:flutter_gen/gen_l10n/l10n.dart';
 
 class AdaptiveScaffoldPrimaryNavigation extends StatelessWidget {
-  final AdaptiveScaffoldShellController controller;
+  final Future<Profile?>? futureProfile;
```
Comment (hoangdat) [inline lib/widgets/layouts/adaptive_layout/adaptive_scaffold_primary_navigation.dart:11]: it is current profile? or profile of who?

**Comment by hoangdat** [comment]
please dont forget to review the case: click on notification

File: lib/pages/chat_list/chat_list_header.dart (hunk 11 above)
Comment (nqhhdev) [reply lib/pages/chat_list/chat_list_header.dart:10]: fixed

File: lib/pages/chat_list/chat_list.dart (hunk 10 above)
Comment (nqhhdev) [reply lib/pages/chat_list/chat_list.dart:66]: fixed

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold.dart (hunk 13 above)
Comment (nqhhdev) [reply lib/widgets/layouts/adaptive_layout/adaptive_scaffold.dart]: Fixed

File: lib/widgets/layouts/adaptive_layout/adaptive_scaffold_primary_navigation.dart (hunk 14 above)
Comment (nqhhdev) [reply lib/widgets/layouts/adaptive_layout/adaptive_scaffold_primary_navigation.dart:11]: Fixed

**Review by hoangdat — APPROVED** [review]

**Pull request merged**